### PR TITLE
[Improvement] add min chunk size

### PR DESCRIPTION
**Pull request by deven298**
As title suggests

### CODE REVIEW

**Review by sidmohanty11 — APPROVED** [review]

File: embedchain/embedchain.py [hunk 1]
```diff
@@ -366,12 +367,14 @@ def _load_and_embed(
...
 
         # Create chunks
-        embeddings_data = chunker.create_chunks(loader, src, app_id=app_id)
+        embeddings_data = chunker.create_chunks(loader, src, app_id=app_id, config=add_config.chunker)
         # spread chunking results
         documents = embeddings_data["documents"]
         metadatas = embeddings_data["metadatas"]
         ids = embeddings_data["ids"]
         new_doc_id = embeddings_data["doc_id"]
+        embeddings = embeddings_data.get("embeddings", None)
```
Comment (deshraj) [inline embedchain/embedchain.py:376]:
```suggestion
        embeddings = embeddings_data.get("embeddings")
```

File: embedchain/config/add_config.py [hunk 2]
```diff
@@ -17,9 +17,15 @@ def __init__(
         chunk_size: Optional[int] = None,
         chunk_overlap: Optional[int] = None,
         length_function: Optional[Callable[[str], int]] = None,
+        min_chunk_size: Optional[int] = None,
```
Comment (deshraj) [inline embedchain/config/add_config.py:20]:
```suggestion
        min_chunk_size: Optional[int] = 0,
```

File: embedchain/config/add_config.py [hunk 3]
```diff
@@ -17,9 +17,15 @@ def __init__(
         chunk_size: Optional[int] = None,
```
Comment (deshraj) [inline embedchain/config/add_config.py:17]:
```suggestion
        chunk_size: Optional[int] = 2000,
```

File: embedchain/config/add_config.py [hunk 4]
```diff
@@ -17,9 +17,15 @@ def __init__(
...
         chunk_overlap: Optional[int] = None,
```
Comment (deshraj) [inline embedchain/config/add_config.py:18]:
```suggestion
        chunk_overlap: Optional[int] = 0,
```

File: embedchain/config/add_config.py [hunk 5]
```diff
@@ -17,9 +17,15 @@ def __init__(
...
         length_function: Optional[Callable[[str], int]] = None,
+        min_chunk_size: Optional[int] = None,
     ):
         self.chunk_size = chunk_size if chunk_size else 2000
-        self.chunk_overlap = chunk_overlap if chunk_overlap else 0
+        self.chunk_overlap = chunk_overlap if chunk_overlap is not None else 100
```
Comment (deshraj) [inline embedchain/config/add_config.py:23]:
```suggestion
        self.chunk_overlap = chunk_overlap
```

File: embedchain/config/add_config.py [hunk 6]
```diff
@@ -17,9 +17,15 @@ def __init__(
...
+        self.min_chunk_size = min_chunk_size if min_chunk_size is not None else 0
```
Comment (deshraj) [inline embedchain/config/add_config.py:24]:
```suggestion
        self.min_chunk_size = min_chunk_size
```

**Review by deshraj — APPROVED** [review]
Generally looks good. Left some minor comments. 

Let's also update the documentation about chunking in a follow up PR.

**Pull request closed without merging**